atlp-rwanda/phantom-be-codebandits
# #CP-17 ch: setup-docker-postgress-database

#### &#x1F534; Check the README file for extensive information &#x1F53D;&#x1F53D;
### What does this PR do?
This PR combines two cards:

- [x] Setup docker
- [x] Setup database with PostgreSQL and typeORM as the ORM
Tasks to be completed:

> [#CP-17](https://trello.com/c/LYn9SiKJ) Setup the project to use PostgreSQL

- [x] TypeORM has been configured.
- [x] The app use postgres database

> [CP-19](https://trello.com/c/EG5zGofX)

- [x] An image of the frontend application has been created
- [x] An image of the backend application has been created
- [x] Database is running in a docker container
- [x] Both docker images can communicate while up and running

### Description of Task to be completed

- [x] Proper documentation has been added to Swagger
- [x] Database is working in docker and outside

### How should this be manually tested?
Detailed instructions can be found in the `README.md` file. Follow the below instructions for PR specifics.
**Configure your environment based on the `.env.sample.md` then follow the following:**
```js
git checkout ch-setup-docker-cp-19
git fetch
npm install
// if you don't have postgres installed, refer to the README for instructions
npm start
```
> **Testing routes and database**
Use the below endpoints to check for the database
```
GET   /api/v1/users   // Get a list of all users
POST   /api/v1/users    //Creating a user with {firstName, lastName, email, password}
```

#### Pre-requisites:

- [x] Clone project
- [x] Downloaded and installed docker
- [x] Have some understanding of some docker commands. Useful ones will be listed below
- [x] Postgres up and running (Optional if you want to use **Docker**)
#### Installation
```js
// clone this PR
git fetch origin
git checkout ch-setup-docker-cp-19
npm install
npm start // <npm run dev>to run in watch mode
```
**With Docker installed**

```
// got to the base directory of the project
docker compose up
```
### Any background context you want to provide?
- When configuring the containers that talk to each other. Use the **container name** in place of host.
- To use the typeORM with **Active Record Pattern** we add target on the Schema
### What are the relevant pivotal tracker stories?
-  [#CP-17](https://trello.com/c/LYn9SiKJ) 
-  [CP-19](https://trello.com/c/EG5zGofX)
### Screenshots (if appropriate)
